src/hooks/resources/device.ts

```diff
@@ -354,6 +354,20 @@ sbvrUtils.addPureHook('PATCH', 'resin', 'device', {
 			);
 		}
 
+		if (request.values.supervisor_version != null) {
+			// When a device checks in with it's initial supervisor version, set the corresponding supervisor_release
+			// resource using its current version
+			waitPromises.push(
+				getCurrentRequestAffectedIds(args).map((id) =>
+					setSupervisorReleaseResource(
+						api,
+						id,
+						request.values.supervisor_version,
+					),
+				),
```

Comment:
This will work but I generally prefer to pass all of the ids in so that the api requests can work on the whole batch at once and have `O(1)` requests rather than `O(n)`

Reply:
I wish you had mentioned that in https://github.com/balena-io/open-balena-api/pull/356#discussion_r433220642. I'm happy to update it if it's preferable, but I'm unclear how this could ever be more than one device at a time unless we start queuing/batching ingesting updates from supervisors?

Reply:
Sorry, when I read only take a single id I saw it as the caller only allowing it to be a single id - eg if it was more it could bail if that should be blocked. Specifically the way I could see a batch being updated would be if it was either a batch of dependent devices being updated or a user doing the update for some reason

Reply:
Just so I understand, dependent devices don't have their own supervisors, correct? This hook will only run once per device, unless the current supervisor has not been properly released and there are no matches (in which case this hook will run at each state endpoint PATCH).

I will rewrite to batch the PATCHes.

Reply:
tl;dr discussed offline, we decided we'll do the GET in one batch and patch device-type by device-type since there's no way to do an inner join in Pine.js.